libraries/hermes/src/classes/TypeRegistry.ts

```diff
@@ -0,0 +1,84 @@
+import ConduitGrpcSdk, { ConduitModel } from '@conduitplatform/grpc-sdk';
+import { isEqual } from 'lodash';
+
+type TransportUpdateHandler = (typeName: string, typeFields: ConduitModel) => void;
+
+// Limitations & Optimizations:
+// - Schema types are not removed from requestedTypes upon deregistration of the routes referencing them
+// - We should ideally be hashing type fields for faster object equality comparison
+
+export class TypeRegistry {
+  private static instance: TypeRegistry;
+  private static transportUpdateHandlers: Map<string, TransportUpdateHandler> = new Map();
+  private readonly schemaTypes: Map<string, ConduitModel> = new Map();
+  private readonly requestedTypes: Set<string> = new Set();
+  private dbAvailable = false;
+
+  private constructor(private readonly grpcSdk: ConduitGrpcSdk) {
+    this.waitForDatabase().then();
```

Comment:
Empty then()

Reply:
That is intentional, we already handle resolution internally. It's just explicit.